doc/source/cluster/kubernetes/k8s-ecosystem/kueue.md

```diff
@@ -0,0 +1,221 @@
+* **LocalQueue**
+  * The LocalQueue `user-queue` is a namespaced object in the `default` namespace which belongs to a ClusterQueue. A typical practice is to assign a namespace to a tenant, team or user, of an organization. Users submit jobs to a LocalQueue, instead of to a ClusterQueue directly.
+* **WorkloadPriorityClass**
+  * The WorkloadPriorityClass `prod-priority` has a higher value than the WorkloadPriorityClass `dev-priority`. This means that RayJob custom resources with the `prod-priority` priority class take precedence over RayJob custom resources with the `dev-priority` priority class.  
+
+Create the Kueue resources:
+```bash
+kubectl apply -f kueue-resources.yaml
+```
+
+## Step 4: Gang scheduling with Kueue
+
+Kueue always admits workloads in “gang” mode.
+In other words, Kueue admits workloads on an “all or nothing” basis, ensuring that no RayJob or RayCluster is partially provisioned.
+Gang scheduling strategy is commonly used to avoid waste of compute resources caused by the naive scheduling of workloads.
+
+Download the RayJob YAML manifest from the KubeRay repository.
+
+```bash
+curl -LO https://raw.githubusercontent.com/ray-project/kuberay/master/ray-operator/config/samples/ray-job.kueue-toy-sample.yaml
+```
+
+Before creating the RayJob, modify the RayJob metadata with:
+
+```yaml
+metadata:
+  generateName: rayjob-sample-
+  labels:
+    kueue.x-k8s.io/queue-name: user-queue
+    kueue.x-k8s.io/priority-class: dev-priority
+```
+
+Create two RayJob custom resources with the same priority `dev-priority`.
+There are some important points to note for the RayJob custom resources:
+* The RayJob custom resource includes 1 head Pod and 1 worker Pod, with each Pod requesting 1 CPU and 2G of memory.
+* The RayJob runs a simple Python script that demonstrates a loop running 600 iterations, printing the iteration number and sleeping for 1 second per iteration. Hence, the RayJob runs for about 600 seconds after the submitting Kubernetes Job starts.
+* The RayJob enables automatic cleanup by setting `shutdownAfterJobFinishes` to true. Therefore, KubeRay deletes the RayCluster after the RayJob finishes.
+  * Kueue doesn't handle the RayJob custom resource with the `shutdownAfterJobFinishes` set to false. See the [limitations of Kueue](https://kueue.sigs.k8s.io/docs/tasks/run_rayjobs/#c-limitations) for more details.
+
+```yaml
+kubectl create -f ray-job.kueue-toy-sample.yaml
+kubectl create -f ray-job.kueue-toy-sample.yaml
+```
+
+Each RayJob custom resource requests 2 CPUs and 4G of memory in total.
+However, the ClusterQueue only has 3 CPUs and 6G of memory in total.
+Therefore, the second RayJob custom resource remains pending, and no Pods from the pending RayJob are created, even though the remaining resources are sufficient for a Pod.
```

Comment:
```suggestion
Therefore, the second RayJob custom resource remains pending, and Kubernetes doesn't create Pods from the pending RayJob, even though the remaining resources are sufficient for a Pod.
```

Reply:
Again, not sure if the subject is Kubernetes, please correct as needed.

Reply:
Update "Kubernetes" to "KubeRay".